scripts/gb-model/retrieve_unavailability_data.py

```diff
@@ -0,0 +1,756 @@
+        logger.info(
+            f"  Total installed capacity: {df_filtered['nominal_power_mw'].sum():.1f} MW"
+        )
+
+    # Business type breakdown
+    if "business_type_desc" in df_filtered.columns:
+        business_breakdown = (
+            df_filtered.groupby("business_type_desc")
+            .agg({"resource_name": "nunique", "resource_mrid": "count"})
+            .round(1)
+        )
+        business_breakdown.columns = ["Unique_Units", "Total_Records"]
+        logger.info(f"  Breakdown by type:\\n{business_breakdown}")
+
+
+def test_api_connection(api_client, zone_code: str) -> bool:
+    """
+    Test API connection with a small date range
+
+    Args:
+        api_client: ENTSOEUnavailabilityAPI instance
+        zone_code: EIC code for bidding zone
+
+    Returns:
+        True if API connection works
+    """
+    try:
+        # Test with a recent date range
+        test_start = datetime.now() - timedelta(days=30)
+        test_end = datetime.now() - timedelta(days=29)
+
+        logger.info(
+            f"Testing API connection with date range: {test_start.date()} to {test_end.date()}"
+        )
+
+        response_content = api_client.get_unavailability_data(
+            bidding_zone=zone_code,
+            period_start=test_start,
+            period_end=test_end,
+            business_type="A53",  # Planned maintenance
+        )
+
+        logger.info(f"Test API response length: {len(response_content)} bytes")
+
+        # Check if it's a ZIP file
+        is_zip = response_content.startswith(b"PK")
+        logger.info(f"Response is ZIP format: {is_zip}")
+
+        return True
+
+    except Exception as e:
+        logger.error(f"API connection test failed: {e}")
+        return False
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from scripts._helpers import mock_snakemake
+
+        snakemake = mock_snakemake("retrieve_unavailability_data")
+
+    configure_logging(snakemake)
+    set_scenario_config(snakemake)
+
+    # Load environment variables from .env file
+    load_dotenv()
+
+    # Get API key from environment variable
+    api_key = os.getenv("ENTSO_E_API_KEY")
+
+    if not api_key:
+        raise ValueError(
+            "ENTSO-E API key not found. Please set ENTSO_E_API_KEY in your .env file or environment variables.\\n"
+            "You can get an API key from: https://transparency.entsoe.eu/usrm/user/createPublicApiUser.do"
+        )
+
+    # Get date parameters from config or use defaults
+    unavailability_config = snakemake.config.get("unavailability", {})
+    start_date = pd.to_datetime(unavailability_config.get("start_date", "2023-01-01"))
+    end_date = pd.to_datetime(unavailability_config.get("end_date", "2023-12-31"))
+
+    bidding_zones = unavailability_config.get("bidding_zones", ["GB"])
+    business_types = unavailability_config.get("business_types", ["planned", "forced"])
+    max_request_days = unavailability_config.get("max_request_days", 7)
```

Comment:
you define these in the snakemake config so you should be pointing to those config values